Portal/src/Datahub.Functions/ConfigureAppService.cs

```diff
@@ -0,0 +1,156 @@
+                    projectAcronym);
+                throw new Exception(
+                    $"App service configuration not found for project acronym {projectAcronym}, could not configure web app");
+            }
+
+            var rgName = appServiceConfiguration.AppServiceRg;
+
+            await ConfigureHttpClient();
+
+            var pipelineUrl = await GetPipelineUrlByName(_config.AdoConfig.AppServiceConfigPipeline);
+            
+            await PostPipelineRun(pipelineUrl, appServiceConfiguration, projectAcronym, rgName);
+        }
+        
+        private async Task ConfigureHttpClient()
+        {
+            var credentials = await _keyVaultService.GetSecret(_config.AdoConfig.PatSecretName);
+            _httpClient = new HttpClient();
+            _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic",
+                Convert.ToBase64String(Encoding.ASCII.GetBytes($":{credentials}")));
+        }
+
+        private async Task<string> GetPipelineUrlByName(string pipelineName)
+        {
+            var url = _config.AdoConfig.GetPipelineURL.Replace("{organization}", _config.AdoConfig.OrgName)
+                .Replace("{project}", _config.AdoConfig.ProjectName);
+            var response = await _httpClient.GetAsync(url);
+            
+            if (!response.IsSuccessStatusCode)
+            {
+                _logger.LogError("Error listing pipelines");
+                throw new Exception($"Error listing pipelines");
+            }
+            
+            var responseContent = await response.Content.ReadAsStringAsync();
+            var pipelines = JsonSerializer.Deserialize<PipelinesResponse>(responseContent)!.Value;
+            var pipeline = pipelines.FirstOrDefault(p => p.Name == pipelineName);
+            
+            if (pipeline is null)
+            {
+                _logger.LogError("Pipeline {PipelineName} not found", pipelineName);
+                throw new Exception($"Pipeline {pipelineName} not found");
+            }
+
+            return pipeline.Url;
+        }
+        
+        private async Task PostPipelineRun(string pipelineUrl, AppServiceConfiguration appServiceConfiguration, string projectAcronym, string rgName)
+        {
+            var body = new
+            {
+                resources = new
+                {
+                    repositories = new
+                    {
+                        self = new
+                        {
+                            refName = "refs/heads/main"
+                        }
+                    }
+                },
+                templateParameters = new
+                {
+                    resourceGroup = rgName,
+                    webAppId = appServiceConfiguration.AppServiceId,
+                    gitUrl = appServiceConfiguration.AppServiceGitRepo,
+                    composePath = appServiceConfiguration.AppServiceComposePath
+                }
+            };
```

Comment:
You're gonna love this snippet I keep saved as a refresher

## Easy way to work with JSON and C#
```csharp
var postdata = new JsonObject
{
    ["moduleList"] = new JsonObject
    {
        ["modules"] = new JsonArray(new JsonObject()
        {
            ["moduleArea"] = "Discovery",
            ["moduleType"] = "ChannelListing",
            ["moduleRequest"] = new JsonObject
            {
                ["consumeRequests"] = new JsonArray(),
                ["resultTemplate"] = "responsive",
                ["alerts"] = new JsonArray(),
                ["profileInfos"] = new JsonArray()
            }
        })
    }
};
```